openhands/integrations/github/github_service.py

```diff
@@ -341,18 +353,65 @@ async def search_repositories(
             # Add is:public to the query to ensure we only search for public repositories
             params['q'] = f'in:name {org}/{repo_name} is:public'
 
-        # Perhaps we should go through all orgs and the search for repos under every org
-        # Currently it will only search user repos, and org repos when '/' is in the name
+        # Handle private repository searches
         if not public and '/' in query:
             org, repo_query = query.split('/', 1)
             query_with_user = f'org:{org} in:name {repo_query}'
             params['q'] = query_with_user
         elif not public:
+            # Expand search scope to include user's repositories and organizations they're a member of
             user = await self.get_user()
-            params['q'] = f'in:name {query} user:{user.login}'
-
+            user_orgs = await self.get_user_organizations()
+
+            # Try a simpler query format: search in user repos and org repos separately
+            all_repos = []
+            
+            # Search in user repositories
+            user_query = f'{query} user:{user.login}'
+            user_params = params.copy()
+            user_params['q'] = user_query
+            logger.info(f'Searching user repos with query: {user_query}')
+            
+            try:
+                user_response, _ = await self._make_request(url, user_params)
+                user_items = user_response.get('items', [])
+                all_repos.extend(user_items)
+                logger.info(f'Found {len(user_items)} repos in user search')
+            except Exception as e:
+                logger.warning(f'User search failed: {e}')
+            
+            # Search in each organization
+            for org in user_orgs:
```

Comment:
Nice! We should also filter against the org name

So if we have a partial match on the org name, get the first 2 repos inside the org


Currently if I type `All-Hands` it will search for the repo `All-Hands` under every org, whereas we could try matching against `All-Hands` org as well and list a couple of repos from there